### PR TITLE
ENH: bfill for interval indexing added GH#51503 #51503 

### LINKED ISSUE #51503: NotImplementedError: method backfill not yet implemented for IntervalIndex
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

```
import pandas as pd
dt = pd.Timestamp.now()
h = pd.Timedelta("1h")

# 3 simple interval ranges, with a big gap between 2nd and 3rd
intervals = pd.IntervalIndex.from_arrays([dt, dt+h, dt+10*h], [dt+h, dt+2*h, dt+11*h], closed="left")

intervals.get_indexer([dt + 5*h], method="bfill")
```
Expectation: the timestamp in `get_indexer` is mapped to the 3rd interval range, but instead this happens:

> NotImplementedError: method backfill not yet implemented for IntervalIndex

### PR DESCRIPTION
**Pull request by hamedgibago**
- [x] closes #51503
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by hamedgibago** [comment]
Test is done for `backfill/bfill` methods with time intervals. `nearest` and `pad/ffill `will be added in future commits.

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -694,13 +694,37 @@ def _get_indexer(
...
             # we should always have self._should_partial_index(target) here
             return self._get_indexer_pointwise(target)[0]
 
         return ensure_platform_int(indexer)
 
+    def _get_indexer_ffill(self, target, method):
+        # GH#51503
+        if method in ("bfill", "backfill"):
+            temp = []
+            for i in range(len(self)):
```
Comment (mroeschke) [inline pandas/core/indexes/interval.py:714]: These for loops are not going to be performant enough for this feature

**Comment by mroeschke** [comment]
Thanks for your pull request @hamedgibago. I noticed this is your first pull request into pandas, so I would recommend first tackling an issue labeled `good first issue`. The issue you are tackling is a bit more complex and has not been triaged by a maintainer yet

**Comment by hamedgibago** [comment]
> 

I'll do `good first issue` issues. Despite this comment you mentioned and not all features added for `bfill`, because All checks have passed and green, will be this PR accepted?

**Comment by mroeschke** [comment]
I am dubious that that for looping solution over each element will be performant enough for this feature

**Comment by hamedgibago** [comment]
> I am dubious that that for looping solution over each element will be performant enough for this feature

I've added this for type of date time check. Let me see if `self._maybe_cast_listlike_indexer` and `self._maybe_convert_i8(t)` will be removed any error happen or not. 
For the second loop of comparing indexes, I'll think another way, as I mentioned in `TODO` comment in the code

**Comment by mroeschke** [comment]
Thanks for this pull request, but it appears there hasn't been much traction on this PR in a while so closing for how. Happy for you to tackle an issue that has been triaged or a `good first issue`

**Pull request closed without merging**